### DISCUSSION TITLE
How to stop charging through API the right way?

### DISCUSSION THREAD

**Discussion by menloperk** [leeyuentuen/alfen_wallbox#71] [General]
How can I safely stop charging (e.g. when sun has set) and start it back up again later (e.g. when sun has risen)? Can I safely put the Socket 1 Operation Mode to In-operative and then later back to Operative? Will it automatically start charging my EV again from there on? Or are there better ways of doing this?

**Comment by leeyuentuen** [comment]
you can try to set the current to 0A? ("wallbox_power_connector_max_current_socket_1" if your wallbox called wallbox)

**Comment by dancit** [comment]
I'have the same need and I made some test using  the **alfen_max_station_current** paramter. setting the state below 6 it forces the wallbox to stop (LoadBalancing Forced OFF status).

while I'm worried about playing with MaxStationCurrent (it should be a fixed value of the fuse protecting the station), i opened a ticket to Alfen support team asking for the right setting. 
the answer was:

> _Currently, there is no "right setting" to use. MaxStationCurrent should be the value of the fuse that is protecting the charging station. Setting this value too high will allow the station to blow the fuse. A better setting to use would be Connector #1 maximum current ("main_normal_max_current_socket_1"). That will allow the station fuse to be protected by the charging station as set in MaxStationCurrent. Because Alfen does not have an official Home Automation interface, there is no other parameter to use for momentary changes of offered current_

unfortunatly this HACS doesn't expose the **main_normal_max_current_socket_1** parameter and the suggested parameter **power_connector_max_current_socket_1** doesn't produce any effect.

any hint?

**Reply by leeyuentuen** [reply]
the main_normal_max_current is "Power Connector Max Current Socket 1"
```
        key="main_normal_max_current_socket_1",
        name="Power Connector Max Current Socket 1",
...
        api_param="2129_0",
```

Also the 6A charging issue, normally the can need '6A' to charge. all the value below 6A will normally not trigger the car to charge

**Reply by dancit** [reply]
thanks @leeyuentuen 

my mistake, I fixed a NodeRed error and now I'm able to modulate the max current using the parameter "power_connector_max_current_socket_1"

**Reply by leeyuentuen** [reply]
@dancit  this discussion can be closed now?

**Reply by dancit** [reply]
yes!

**Comment by menloperk** [comment]
This ("Power Connector Max Current Socket 1") is what I have been using for a couple of days now and it seems to work fine for me. I switch between 0A and 16A based on my schedule... my EV seems to restart charging just fine when I switch it back to 16A (like when my solar panels start producing again in the morning).
